### PR TITLE
Authentication section

### PR DESCRIPTION
**Pull request by rAzizbekyan**

### CODE REVIEW

File: src/components/Form.svelte [hunk 1]
```diff
@@ -116,14 +105,43 @@
...
+        {/if}
+      {:else if $data.authMethod === 'SASL_PLAINTEXT'}
+        <TextField
+          name="saslJaasConfig"
+          label="sasl.jaas.config"
+          errors={$errors.saslJaasConfig}
+          required
+        />
+      {:else if $data.authMethod === 'SASL_SSL'}
+        <CheckboxField name="securedWithSSL" label="Secured with SSL" />
```
Comment (workshur) [inline src/components/Form.svelte]:
```suggestion
      {:else if $data.authMethod === 'SASL_SSL'}
        
```

$data.authMethod === 'SASL_SSL' means that securedWithSSL checked by default, and user have no ability to uncheck it.

File: src/components/Label.svelte [hunk 2]
```diff
@@ -1,7 +1,20 @@
 <script type="ts">
   export let forTarget: string;
+  export let required = false;
 </script>
 
-<label class="block text-sm font-medium text-gray-700 whitespace-nowrap mr-2" for={forTarget}>
+<label
+  class="{required
+    ? 'required'
+    : ''} block text-sm font-medium text-gray-700 whitespace-nowrap mr-2"
```
Comment (workshur) [inline src/components/Label.svelte]:
```suggestion
  class="block text-sm font-medium text-gray-700 whitespace-nowrap mr-2"
  class:required={required}
```

File: src/lib/clusterConfigurationSchema.ts [hunk 3]
```diff
@@ -52,15 +74,14 @@ const clusterConfigurationSchema = object({
   saslMechanism: string()
     .label('sasl_mechanism')
     .when('authMethod', {
-      is: 'SASL_PLAINTEXT',
+      is: 'SASL_SSL',
       then: (s) =>
         s.required().oneOf(['PLAIN', 'AWS_MSK_IAM', 'SCRAM-SHA-256', 'SCRAM-SHA-512', 'GSSAPI'])
```
Comment (workshur) [inline src/lib/clusterConfigurationSchema.ts]:
```suggestion
        s.required().oneOf(['AWS_MSK_IAM', 'SCRAM-SHA-256', 'SCRAM-SHA-512', 'GSSAPI'])
```

File: src/components/Form.svelte (hunk 1 above)
Comment (rAzizbekyan) [reply src/components/Form.svelte]: done

File: src/components/Label.svelte (hunk 2 above)
Comment (rAzizbekyan) [reply src/components/Label.svelte]: done

File: src/lib/clusterConfigurationSchema.ts (hunk 3 above)
Comment (rAzizbekyan) [reply src/lib/clusterConfigurationSchema.ts]: done

File: src/components/Form.svelte [hunk 4]
```diff
@@ -116,36 +107,58 @@
...
         <SelectField
           name="saslMechanism"
           label="SASL Mechanism"
           containerClass="col-span-6"
-          options={['PLAIN', 'AWS_MSK_IAM', 'SCRAM-SHA-256', 'SCRAM-SHA-512', 'GSSAPI']}
+          options={['AWS_MSK_IAM', 'SCRAM-SHA-256', 'SCRAM-SHA-512', 'GSSAPI']}
         />
-
-        {#if $data.saslMechanism === 'GSSAPI'}
+        {#if $data.saslMechanism === 'SCRAM-SHA-256' || $data.saslMechanism === 'SCRAM-SHA-512'}
```
Comment (workshur) [inline src/components/Form.svelte]: pls add GSSAPI too

Comment (workshur) [reply src/components/Form.svelte]:
```ts
['GSSAPI', 'SCRAM-SHA-256', 'SCRAM-SHA-512'].includes($data.saslMechanism)
```

File: src/lib/clusterConfigurationSchema.ts [hunk 5]
```diff
@@ -79,8 +106,9 @@ const clusterConfigurationSchema = object({
       is: 'GSSAPI',
       then: (s) => s.required()
     }),
-  saslSSL: object().when('authMethod', {
-    is: 'SASL_SSL',
+  saslSSL: object().when(['authMethod', 'jmxSslEnabled'], {
```
Comment (workshur) [inline src/lib/clusterConfigurationSchema.ts]: Pls talk with @Haarolean. I'm not sure we use same CA for both

**Review by workshur — COMMENTED** [review]
looks good so far

File: src/lib/clusterConfigurationSchema.ts (hunk 5 above)
Comment (Haarolean) [reply src/lib/clusterConfigurationSchema.ts]: Yep, ssl for jmx can be enabled with any type of authentication. That's unrelated in this case.

File: src/components/Form.svelte (hunk 4 above)
Comment (rAzizbekyan) [reply src/components/Form.svelte]: done

File: src/lib/clusterConfigurationSchema.ts (hunk 5 above)
Comment (rAzizbekyan) [reply src/lib/clusterConfigurationSchema.ts]: ok

**Review by workshur — APPROVED** [review]

**Pull request merged**